clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchUtil.scala

```diff
@@ -56,5 +56,18 @@ object ElasticsearchUtil {
       */
     def unsafeGet[T: Decoder](key: String): T =
       json.hcursor.get[T](key).fold(throw _, identity)
+
+    def getAsString(key: String): String = {
+      json.asObject
+        .flatMap(_.apply(key))
+        .map {
+          case s if s.isString =>
+            s.asString.getOrElse("")
+          case a if a.isArray =>
```

Comment:
I think we should explode on an array / object field, it doesn't make sense to have it as part of the key.

Reply:
We shouldn't be using this method outside of ID generation (now that I'm re-looking at it, I'd even rather move it to ElasticsearchIndex as a private method there). The whole setup of concatenating pieces with `.` and not double-quoting strings is designed to fit the ID-gen use case, and doesn't make sense in other contexts.

Reply:
that makes sense. I'll move it into EsI.